qiita_db/artifact.py

```diff
@@ -615,12 +624,15 @@ def delete(cls, artifact_id):
                 qdb.util.move_filepaths_to_upload_folder(
                     study.id, filepaths)
 
-                sql = """UPDATE qiita.prep_template
-                         SET artifact_id = NULL
-                         WHERE prep_template_id IN %s"""
-                qdb.sql_connection.TRN.add(
-                    sql, [tuple(pt.id for a in all_artifacts
-                                for pt in a.prep_templates)])
+                # there are cases that an artifact would not be linked to a
```

Comment:
Now we have artifacts (like those generated by the sequencing processing pipeline) without prep/analysis linkages.